addon/components/materialize-carousel.js

```diff
@@ -0,0 +1,29 @@
+import Ember from 'ember';
+import layout from '../templates/components/materialize-carousel';
+
+export default Ember.Component.extend({
+  layout,
+  classNames: ['carousel'],
+
+  didRender (attrs) {
+    this._super (...arguments);
+
```

Comment:
The `if` block below is not necessary because it does not matter if they attributes are set. If you get the attribute and it is not present, then it will return undefined. If all are undefined, then there are no options set. Based on your logic below, the code in the `if` statement is executed if one attribute is set. So, there is case where all will be undefined in the options except for one.

Lastly, the `carousel()` method will execute only if you set at least 1 options. This should not be the case. It needs to be executed the first time the element is inserted into the DOM, and each time one of the attributes is updated. 

Right now, the carousel will not render with it is first created.